src/host/selectionState.cpp

```diff
@@ -197,6 +197,38 @@ til::point Selection::GetSelectionAnchor() const noexcept
     return _coordSelectionAnchor;
 }
 
+// Routine Description:
+// - Gets the current selection begin and end anchor positions.
+// Arguments:
+// - ordered - if true, arrange the selection anchors such that the first anchor
+//             is at the top left corner, and the second one is at the bottom
+//             right of the selection area
+// Return Value:
+// - The current selection anchors
+std::pair<til::point, til::point> Selection::GetSelectionAnchors(const bool ordered) const noexcept
+{
+    if (!_fSelectionVisible)
+    {
+        // return anchors that represent an empty selection
+        return { { 0, 0 }, { -1, -1 } };
+    }
+
+    auto startSelectionAnchor = _coordSelectionAnchor;
+
+    // _coordSelectionAnchor is at one of the corners of _srSelectionRects
+    // endSelectionAnchor is at the exact opposite corner
+    til::point endSelectionAnchor;
+    endSelectionAnchor.x = (_coordSelectionAnchor.x == _srSelectionRect.left) ? _srSelectionRect.right : _srSelectionRect.left;
+    endSelectionAnchor.y = (_coordSelectionAnchor.y == _srSelectionRect.top) ? _srSelectionRect.bottom : _srSelectionRect.top;
+
+    if (ordered && startSelectionAnchor > endSelectionAnchor)
+    {
+        std::swap(startSelectionAnchor, endSelectionAnchor);
+    }

```

Comment:
Does anyone ever want them unordered? 🤔 I think this could just be the default.

Reply:
Yes, we can make this the default thing we do when returning selection anchors. If in the future a consumer wants to see the anchors in the _original_ order, we can add the flag again 👍